### PR TITLE
feat(tui): preserve cursor position after notification actions

### PR DESCRIPTION
**Pull request by cristianoliveira**
## Summary
Fixes the UX issue where the cursor would jump to the top of the list after performing actions (dismiss, mark as read, mark as unread). The cursor now stays in the same position or moves to the next notification when the current one is dismissed.

## Problem Description
**Bug:** When a user performs any action (dismiss, mark as read, mark as unread), the cursor jumps back to the top of the list (position 0).

**Expected Behavior:** Cursor should stay in the same position or move to the next notification when the current one is dismissed.

## Root Cause
The `applySearchFilter()` function unconditionally set `m.cursor = 0` at the end. This function was called by `loadNotifications()` after every action, causing the cursor to reset to the top each time.

## Solution Implemented

**1. Added `preserveCursor` parameter to filter functions:**
- `applySearchFilter(preserveCursor bool)` – Only resets cursor when `preserveCursor = false`
- `loadNotifications(preserveCursor bool)` – Propagates the parameter

**2. Modified action handlers to preserve cursor:**
- `handleDismiss()` – Captures cursor index before action, preserves at same position after reload
- `markSelectedRead()` – Captures notification ID, restores cursor to same notification
- `markSelectedUnread()` – Captures notification ID, restores cursor to same notification

**3. Added helper functions for cursor management:**
- `getNodeIdentifier()` – Returns stable node identifier
- `findNodeByIdentifier()` – Finds node by identifier in visible nodes
- `restoreCursor()` – Restores cursor to saved position
- `adjustCursorBounds()` – Ensures cursor is within valid bounds

**4. Updated all call sites (14 total):**
- Search/filter operations: Pass `false` (reset cursor – expected behavior)
- Action handlers: Pass `true` (preserve cursor – new behavior)

**5. Added comprehensive test coverage (8 new tests):**
- Dismiss in flat view
- Dismiss in grouped view
- Mark as read
- Mark as unread
- Edge cases (last item, single item)
- Helper function tests

## Modified Files
- `internal/tui/state/model.go` – Core implementation
- `internal/tui/state/model_test.go` – Tests updated and added
- `internal/tui/state/model_bench_test.go` – Benchmarks updated

## Validation
- ✅ All tests pass (Go + Bats)
- ✅ All linting passes
- ✅ Code is formatted
- ✅ Build succeeds
- ✅ Works with SQLite backend
- ✅ Compatible with performance optimizations

## Behavior Changes
**Before:** Cursor jumps to top after every action, forcing users to re‑navigate.
**After:** Cursor stays in place (or moves to next notification if current is dismissed), providing a smooth, predictable UX.

## Commit Included
This PR includes commit `c692963` (feat(tui): add cursor preservation for notification actions) merged with latest `main` branch changes.

### CODE REVIEW

File: internal/tui/state/model.go [hunk 1]
```diff
@@ -459,10 +460,89 @@ func (m *Model) applySearchFilter() {
 		m.invalidateCache()
 		m.visibleNodes = nil
 	}
-	m.cursor = 0
+	if !preserveCursor {
+		m.cursor = 0
+	}
```
Comment (cristianoliveira) [inline internal/tui/state/model.go]: Do we even need this? Can't we have a function that restore the cursor if needed?
